tests/bears/BearTest.py

```diff
@@ -262,6 +270,35 @@ def test_new_result(self):
         expected = Result.from_values(bear, 'test message', '/tmp/testy')
         self.assertEqual(result, expected)
 
+
+class BearDownloadTest(BearTestBase):
+
+    def setUp(self):
+        super().setUp()
+        self.mock_url = 'https://test.com'
+        self.filename = 'test.html'
+
+    def test_connection_timeout_mocked(self):
+        exc = requests.exceptions.ConnectTimeout
+        with requests_mock.Mocker() as reqmock:
+            reqmock.get(self.mock_url, exc=exc)
+            with pytest.raises(exc):
+                self.uut.download_cached_file(
+                    self.mock_url, self.filename)
+
+    def test_connection_failure(self):
+        exc = requests.exceptions.HTTPError
+        with pytest.raises(exc):
+            self.uut.download_cached_file(
+                'http://httpbin.org/status/418', self.filename)
+
+    def test_read_failure_httpbin(self):
+        exc = requests.exceptions.HTTPError
+        with pytest.raises(exc):
+            self.uut.download_cached_file(
+                'http://httpbin.org/drip?duration=5&numbytes=5&code=500',
+                self.filename)
+
```

Comment:
This doesnt need to check all status.  While it is using a real server, it is still a very fake test covering only a basic scenario.  The same basic scenario with different codes is not worth testing.

Perhaps a useful addition is verifying that the status code `500` does appear in the exception message.  I'll add that.